source/MaterialXContrib/MaterialXNode/MaterialXData.cpp

```diff
@@ -96,64 +103,124 @@ void MaterialXData::generateXML()
+    if (shader)
     {
-        if (MHWRender::MFragmentManager* fragmentMgr = theRenderer->getFragmentManager())
+        std::ostringstream stream;
+        // Note: This name must match the the fragment name used for registration
+        // or the registration will fail.
+        _generator.generate(_fragmentName, shader.get(), nullptr, stream);
+        _fragmentWrapper = stream.str();
+        if (_fragmentWrapper.empty())
+        {
+            return;
+        }
+        else
+        {
+            // Strip out any '\r' characters.
+            _fragmentWrapper.erase(std::remove(_fragmentWrapper.begin(), _fragmentWrapper.end(), '\r'), _fragmentWrapper.end());
+        }
+
+        // Extract out the input fragment parameter names along with their
+        // associated Element paths to allow for value binding.
+        const mx::ShaderStage& ps = shader->getStage(mx::Stage::PIXEL);
+        for (auto uniformsIt : ps.getUniformBlocks())
         {
-            const bool fragmentExists = (getFragmentName().size() > 0)
-                && fragmentMgr->hasFragment(getFragmentName().c_str());
+            const mx::VariableBlock& uniforms = *uniformsIt.second;
 
-            if (!fragmentExists)
+            // Skip light uniforms
+            if (uniforms.getName() == mx::HW::LIGHT_DATA)
             {
-                // XML should come from here. For now allow to get from a input path
-                // std::stringstream glslStream;
-                // getXML(glslStream);
-                std::string xmlFileName(Plugin::instance().getResourcePath() / ogsXmlPath);
-
-                // TODO: This should not be hard-coded
-                std::string dumpPath("d:/work/shader_dump/");
-                MaterialX::FileSearchPath path = MaterialX::getEnvironmentPath("TEMP");
-                if (path.size() > 0)
-                {
-                    dumpPath = path[0].asString();
-                }
-                fragmentMgr->setEffectOutputDirectory(dumpPath.c_str());
-                fragmentMgr->setIntermediateGraphOutputDirectory(dumpPath.c_str());
-                MString fragmentName = fragmentMgr->addShadeFragmentFromFile(xmlFileName.c_str(), false);
+                continue;
+            }
 
-                if (fragmentName.length() == 0)
+            for (size_t i = 0; i < uniforms.size(); i++)
+            {
+                const mx::ShaderPort* port = uniforms[i];
+                const std::string& path = port->getPath();
+                if (!path.empty())
                 {
-                    throw MaterialX::Exception("Failed to add OGS shader fragment from file.");
+                    std::string name = port->getVariable();
+                    if (port->getType()->getSemantic() == mx::TypeDesc::SEMANTIC_FILENAME)
+                    {
+                        // Strip out the "sampler" post-fix to get the texture name.
+                        size_t pos = name.find(mx::OgsXmlGenerator::SAMPLER_SUFFIX);
```

Comment:
We generate <filename>Sampler in the code so the variable has this name. We want back the "root" which is the name with "Sampler" stripped away as that is the Shader instance parameter name we're interested in.